bricks/test_optimizer/hooks/lib/pre_gen.dart

```diff
@@ -28,21 +28,55 @@ Future<void> run(HookContext context) async {
   final flutterSdkRegExp = RegExp(r'sdk:\s*flutter$', multiLine: true);
   final isFlutter = flutterSdkRegExp.hasMatch(pubspecContents);
 
-  final tests = testDir
-      .listSync(recursive: true)
-      .where((entity) => entity.isTest)
-      .map(
-        (entity) => path
-            .relative(entity.path, from: testDir.path)
-            .replaceAll(r'\', '/'),
-      )
-      .toList();
-
-  context.vars = {'tests': tests, 'isFlutter': isFlutter};
+  final identifierGenerator = StringIdGenerator();
+  final testIdentifierTable = <Map<String, String>>[];
```

Comment:
nit: I would rename this to `testsNormalized`, something a bit descriptive on the why

Reply:
I think testIdentifierTable is descriptive. Reading as "matches a test file to its dart identifier".

I like to include the "table" in the variable to resemble common computer science terms such as finger tables as in a Distributed Hash Table (DHT), or symbol tables in the compile process.

I updated the README example to give some intuition on why this is done.